event/views.py

```diff
@@ -1,3 +1,77 @@
-from django.shortcuts import render
+from django.shortcuts import get_object_or_404
+from django.http import JsonResponse
+from django.contrib.auth.mixins import LoginRequiredMixin
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAdminUser
+from .models import Event
+import json
 
-# Create your views here.
+class EventListView(APIView):
+    """
+    API view to retrieve a list of all events.
+    """
+    def get(self, request):
+        event = Event.objects.all()
```

Comment:
We shouldn't fetch all events, it should be specific to a user concern.

Reply:
Noted.